test/config/config.test.ts

```diff
@@ -125,34 +157,166 @@ describe('Config', () => {
+      if (pjson) test = test.stub(util, 'loadJSON', () => Promise.resolve(pjson))
+      test = test.add('config', async () => {
+        const config = await Config.load()
+        config.plugins = plugins
+        config.pjson.oclif.plugins = ['@My/pluginb', '@My/plugina']
+        config.pjson.dependencies = {'@My/pluginb': '0.0.0', '@My/plugina': '0.0.0'}
+        return config
+      })
+      // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+      // @ts-ignore
+      return {
+        it(expectation: string, fn: (config: Interfaces.Config) => any) {
+          test
+          .do(({config}) => fn(config))
+          .it(expectation)
+          return this
+        },
+      }
+    }
+
+    findCommandTestConfig()
+    .it('find command with no duplicates', config => {
+      expect(config.findCommand('foo:bar', {must: true}))
+    })
+    findCommandTestConfig({commandIds: ['foo:bar', 'foo:bar']})
+    .it('find command with duplicates', config => {
+      expect(config.findCommand('foo:bar', {must: true}))
+    })
+    findCommandTestConfig({types: ['core', 'user']})
+    .it('find command with no duplicates core/user', config => {
+      const command = config.findCommand('foo:bar', {must: true})
+      expect(command).to.have.property('id', 'foo:bar')
+      expect(command).to.have.property('pluginType', 'core')
+      expect(command).to.have.property('pluginAlias', '@My/plugina')
+    })
+    findCommandTestConfig({types: ['user', 'core']})
+    .it('find command with no duplicates user/core', config => {
+      const command = config.findCommand('foo:bar', {must: true})
+      expect(command).to.have.property('id', 'foo:bar')
+      expect(command).to.have.property('pluginType', 'user')
+    })
+    findCommandTestConfig({commandIds: ['foo:bar', 'foo:bar'], types: ['core', 'user']})
+    .it('find command with duplicates core/user', config => {
+      const command = config.findCommand('foo:bar', {must: true})
+      expect(command).to.have.property('id', 'foo:bar')
+      expect(command).to.have.property('pluginType', 'core')
+      expect(command).to.have.property('pluginAlias', '@My/plugina')
+    })
+    findCommandTestConfig({commandIds: ['foo:bar', 'foo:bar'], types: ['user', 'core']})
+    .it('find command with duplicates user/core', config => {
+      const command = config.findCommand('foo:bar', {must: true})
+      expect(command).to.have.property('id', 'foo:bar')
+      expect(command).to.have.property('pluginType', 'core')
+      expect(command).to.have.property('pluginAlias', '@My/pluginb')
+    })
+    findCommandTestConfig({commandIds: ['foo:bar', 'foo:bar'], types: ['user', 'user']})
+    .it('find command with duplicates user/user', config => {
+      const command = config.findCommand('foo:bar', {must: true})
+      expect(command).to.have.property('id', 'foo:bar')
+      expect(command).to.have.property('pluginType', 'user')
+      expect(command).to.have.property('pluginAlias', '@My/plugina')
+    })
```

Comment:
Not sure that this is a good or needed test because the new `findCommand` function doesn't make any guarantees when you ask for a duplicate command from `user` type plugins, right?

This is the kind of thing that we'd actually need a DAG for.

Reply:
I'm fine to keep it but I don't think we'll be able to say that `sf` will return any particular command in this scenario.

Reply:
I agree that this is an indeterminate use case, but I like having the test in place in case.